dask/tests/test_callbacks.py

```diff
@@ -12,3 +12,15 @@ def _start(self, dsk):
         get_sync({'x': 1}, 'x')
 
     assert flag[0] is True
+
+def test_start_state_callback():
+    flag = [False]
+
+    class MyCallback(Callback):
+        def _start_state(self, dsk, state):
+            flag[0] = True
```

Comment:
Perhaps test something with the arguments? Currently this would still pass even if the arguments to the callback are accidentally swapped in the scheduler.